devito/arch/compiler.py

```diff
@@ -501,6 +501,22 @@ def __lookup_cmds__(self):
         self.MPICXX = 'mpicxx'
 
 
+class ICXCompiler(Compiler):
```

Comment:
Shouldn't this be an IntelCompiler

Reply:
Did you try it? How does it compare with ICC in perf?

Reply:
I wrote it in a similar way to DPCPP. So as not inheriting from Intel.

Did not see any perf advantage on a few acoustic/tti problems locally. 
icc is gonna be deprecated soon and icx will be the default from Intel

Reply:
I agree that not inheriting from either IntelCompiler or ClangCompiler is a bit odd, but I can live with it for now, I think...

Reply:
Conceptually I would agree, but there is not much overlap in flags etc...